src/Gui/CommandDoc.cpp

```diff
@@ -1676,7 +1677,19 @@ class StdCmdExpression : public Gui::Command
 
             size_t len = (found?m2[0].first:tend) - m[0].second;
             try {
-                App::ExpressionPtr expr(App::Expression::parse(obj,std::string(m[0].second,len)));
+                // Check if the expression body contains a single
+                // non-whitespace character '#'.  This is used to signal the
+                // user's intention of unbinding the property.
+                bool empty = false;
+                const char *t = m[0].second;
+                for(;*t && std::isspace((int)*t);++t);
+                if(*t == '#') {
+                    for(++t;*t && std::isspace((int)*t);++t);
+                    empty = !*t;
+                }

```

Comment:
I think using a regular expression here would better show the intent of the code, and would be easier for later readers to follow. Something like (untested):
```
boost::regex re("^\s*#\s*$"); // A line consisting of only whitespace, a #, and more whitespace
bool empty = boost::regex_match(t, re);
```
(or you could use std::regex instead)